core/domain/collection_services.py

```diff
@@ -154,7 +184,18 @@ def get_collection_summary_from_model(collection_summary_model):
 
 
 def get_collection_by_id(collection_id, strict=True, version=None):
-    """Returns a domain object representing a collection."""
+    """Returns a domain object representing a collection.
+
+    Args:
+        collection_id: str. ID of the collection.
+        strict: bool. Indicates whether to fail noisily if no
+            collection with the given id exists in the datastore.
+        version: str. The version number of the collection to be retrieved.
```

Comment:
Sorry I'm not sure what `None` means though, does it get the latest version of the collection? The super method `get(cls, entity_id, strict=True)` of `BaseModel` gets called in the case that `None` is passed.